modules/highgui/CMakeLists.txt

```diff
@@ -1,26 +1,5 @@
 set(the_description "High-level GUI")
 
-set(ENABLE_PLUGINS_DEFAULT ON)
-if(EMSCRIPTEN OR IOS OR WINRT)
-  set(ENABLE_PLUGINS_DEFAULT OFF)
-endif()
-set(HIGHGUI_PLUGIN_LIST "" CACHE STRING "List of GUI backends to be compiled as plugins (gtk, gtk2/gtk3, qt, win32 or special value 'all')")
-set(HIGHGUI_ENABLE_PLUGINS "${ENABLE_PLUGINS_DEFAULT}" CACHE BOOL "Allow building and using of GUI plugins")
-mark_as_advanced(HIGHGUI_PLUGIN_LIST HIGHGUI_ENABLE_PLUGINS)
```

Comment:
options are moved to `init.cmake` to use them there.

We don't need to detect all possible variants of 3rdparty backends if we don't have enabled plugins at all (e.g., line 13 of `detect_gtk.cmake`).